solutionbox/structured_data/datalab_solutions/structured_data/_package.py

```diff
@@ -0,0 +1,236 @@
+    transforms_config_file: File path to the config file.
+    train_percent: Int in range [0, 100].
+    eval_percent: Int in range [0, 100].
+    train_percent: Int in range [0, 100].
+    project_id: String. The GCE project to use. Defaults to the notebook's
+        default project id.
+    job_name: String. Job name as listed on the Dataflow service. If None, a 
+        default job name is selected.
+  """
+  _check_transforms_config_file(transforms_config_file)
+
+  percent_flags = _percent_flags(train_percent, eval_percent, test_percent)
+  this_folder = os.path.dirname(os.path.abspath(__file__))
+  project_id = project_id or _default_project()
+  job_name = job_name or ('structured-data-' + 
+                          datetime.datetime.now().strftime('%Y%m%d%H%M%S'))
+
+  cmd = ['python',
+         os.path.join(this_folder, 'preprocess/preprocess.py'),
+         '--cloud',
+         '--project_id=%s' % project_id,
+         '--job_name=%s' % job_name,
+         '--input_file_path=%s' % input_file_path,
+         '--output_dir=%s' % output_dir,
+         '--transforms_config_file=%s' % transforms_config_file] + percent_flags
+  
+  print('Cloud preprocess, running command: %s' % ' '.join(cmd))
+  _run_cmd(' '.join(cmd))
+
+  print 'Cloud preprocessing job submitted.'
+
+  if (_is_in_IPython()):
+    import IPython
+    
+    dataflow_url = ('https://console.developers.google.com/dataflow?project=%s' 
+                    % project_id)
+    html = ('<p>Click <a href="%s" target="_blank">here</a> to track '
+            'preprocessing job.</p><br/>' % dataflow_url)
+    IPython.display.display_html(html, raw=True)
+
+
+
+def local_train(preprocessed_dir, transforms_config_file, output_dir,
+                layer_sizes=None, max_steps=None):
+  """Train model locally.
+  Args:
+    preprocessed_dir: The output directory from preprocessing. Must contain 
+        files named features_train*.tfrecord.gz, features_eval*.tfrecord.gz,
+        and metadata.json. Can be local or GCS path.
+    transforms_config_file: File path to the config file.
+    output_dir: Output directory of training.
+    layer_sizes: String. Represents the layers in the connected DNN. 
+        If the model type is DNN, this must be set. Example "10 3 2", this will
+        create three DNN layers where the first layer will have 10 nodes, the
+        middle layer will have 3 nodes, and the laster layer will have 2 nodes.
+    max_steps: Int. Number of training steps to perform.
+  """
+  
+  #TODO(brandondutra): fix to TF 0.12
+  #TODO(brandondutra): allow other flags to be set.
+
+  train_filename = os.path.join(preprocessed_dir, 'features_train*')
+  eval_filename = os.path.join(preprocessed_dir, 'features_eval*')
+  metadata_filename = os.path.join(preprocessed_dir, 'metadata.json')
+  this_folder = os.path.dirname(os.path.abspath(__file__))
+  cmd = ['cd %s &&' % this_folder, 'gcloud alpha ml local train',
```

Comment:
Side question: could reset session also kill any running child process? That sounds like something useful others would need too.

Are you recommending something like this:
local_preprocess: call code directly 
cloud_preprocess: call code directly 
local_train: call code directly 
cloud_train: use datalab/mlalpha/_cloud_runner.py
local_predict: call code directly
cloud_predict: use datalab/mlalpha/_cloud_predictor.py
local_batch_predict: call code directly
cloud_batch_predict: Does datalab support this? Is the only option gcloud?